plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/coupon/test/index.tsx

```diff
@@ -12,12 +12,15 @@ import { VALIDATION_STORE_KEY } from '@woocommerce/block-data';
 import { TotalsCoupon } from '..';
 
 describe( 'TotalsCoupon', () => {
-	it( "Shows a validation error when one is in the wc/store/validation data store and doesn't show one when there isn't", () => {
+	it( "Shows a validation error when one is in the wc/store/validation data store and doesn't show one when there isn't", async () => {
+		const user = userEvent.setup();
 		const { rerender } = render( <TotalsCoupon instanceId={ 'coupon' } /> );
 
 		const openCouponFormButton = screen.getByText( 'Add a coupon' );
 		expect( openCouponFormButton ).toBeInTheDocument();
-		userEvent.click( openCouponFormButton );
+		await act( async () => {
+			await user.click( openCouponFormButton );
```

Comment:
I don't feel great having to wrap actions in act. Supposedly testing library already does that, but for whatever reason, we still get act warnings, wrapping actions in act fixes the issue. In future we should explore why this happens with our components and see if we can fix this.